cmd/tailscale/ui.go

```diff
@@ -509,86 +620,191 @@ func (ui *UI) menuClicked(btn *widget.Clickable) bool {
-						layout.Rigid(func(gtx C) D {
+		return layout.Center.Layout(gtx, func(gtx C) D {
+			gtx.Constraints.Min.X = gtx.Px(unit.Dp(250))
+			gtx.Constraints.Max.X = gtx.Constraints.Min.X
+			return layoutDialog(gtx, func(gtx C) D {
+				return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						// Header.
+						return layout.Inset{
+							Top:   unit.Dp(16),
+							Right: unit.Dp(20),
+							Left:  unit.Dp(20),
+						}.Layout(gtx, func(gtx C) D {
+							l := material.Body1(ui.theme, "Route traffic via...")
+							l.Font.Weight = text.Bold
+							return l.Layout(gtx)
+						})
+					}),
+					layout.Flexed(1, func(gtx C) D {
+						gtx.Constraints.Min.Y = 0
+						// Add "none" exit node.
+						n := len(exits) + 1
+						return d.list.Layout(gtx, n, func(gtx C, idx int) D {
+							switch idx {
+							case n - 1:
+							}
+							node := ExitNode{Label: "None", Online: true}
+							if idx >= 1 {
+								node = exits[idx-1]
+							}
+							lbl := node.Label
+							if !node.Online {
+								lbl = lbl + " (offline)"
+							}
+							btn := material.RadioButton(ui.theme, &d.exits, string(node.ID), lbl)
+							if !node.Online {
+								btn.Color = rgb(0xbbbbbb)
+								btn.IconColor = btn.Color
+							}
+							return layout.Inset{
+								Top:   unit.Dp(16),
+								Right: unit.Dp(16),
+								Left:  unit.Dp(16),
+							}.Layout(gtx, btn.Layout)
+						})
+					}),
+					layout.Rigid(func(gtx C) D {
+						// Done button.
+						return layout.E.Layout(gtx, func(gtx C) D {
 							return layout.Inset{
-								Top:    unit.Dp(16),
-								Right:  unit.Dp(16),
-								Left:   unit.Dp(16),
+								Left:   unit.Dp(4),
+								Right:  unit.Dp(4),
 								Bottom: unit.Dp(4),
 							}.Layout(gtx, func(gtx C) D {
-								gtx.Constraints.Min.X = minWidth
-								var expiryStr string
-								const fmtStr = time.Stamp
-								switch {
-								case expiry.IsZero():
-									expiryStr = "Expires: (never)"
-								case time.Now().After(expiry):
-									expiryStr = fmt.Sprintf("Expired: %s", expiry.Format(fmtStr))
-								default:
-									expiryStr = fmt.Sprintf("Expires: %s", expiry.Format(fmtStr))
-								}
-								l := material.Caption(ui.theme, expiryStr)
-								l.Color = rgb(0x8f8f8f)
-								dims := l.Layout(gtx)
-								if w := dims.Size.X; w > maxWidth {
-									maxWidth = w
-								}
-								return dims
-							})
-						}),
-					}
-					for i := 0; i < len(items); i++ {
-						it := &items[i]
-						children = append(children, layout.Rigid(func(gtx C) D {
-							return material.Clickable(gtx, it.btn, func(gtx C) D {
-								return layout.UniformInset(unit.Dp(16)).Layout(gtx, func(gtx C) D {
-									gtx.Constraints.Min.X = minWidth
-									dims := material.Body1(ui.theme, it.title).Layout(gtx)
-									if w := dims.Size.X; w > maxWidth {
-										maxWidth = w
-									}
-									return dims
+								return material.Clickable(gtx, &d.done, func(gtx C) D {
```

Comment:
Let's update this dialog so tap an option immediately selects it. This also means we can do away with the "Done" button. It looks like Android handles overflow scrolling on these dialogs well by default, but if we need space to close the dialog we can add an explicit "Cancel" button too.

Here's a reference from the stock messages app on my Motorola G5.

https://user-images.githubusercontent.com/303731/110039551-68226c80-7d0f-11eb-93fd-aec5859d28cd.mp4

Open to other approaches if you think there's a more natural Android pattern here.